### PR TITLE
Observability as Code Scenarios

### PR DESCRIPTION
**Pull request by lakshaykaushik**
This aims to close https://github.com/microsoft/code-with-engineering-playbook/issues/457

### CODE REVIEW

File: observability/readme.md [hunk 1]
```diff
@@ -62,6 +62,14 @@ contributions.
 
 Read more [here](pitfalls.md) to understand what to watch out for while designing and building an observable system.
 
+## Examples of Observability As Code
+
+1. Dashboards as Code- You can create Monitoring Dashboards as Json or XML templates, such templates can be checked into the code repository for source control, any change in the dashboards can be reviewed and automation can be built for enabling the dashboard. [More about how to do this in Azure A](https://docs.microsoft.com/en-us/azure/azure-portal/azure-portal-dashboards-create-programmatically).
```
Comment (Guilhermeslucas) [inline observability/readme.md:67]: Should we also add some reference to grafana dashboards, if any? I am asking about it cause we are mentioning grafana all over the observability section,

File: observability/readme.md [hunk 2]
```diff
@@ -62,6 +62,14 @@ contributions.
...
+
+2. Alerts as Code- Alerts can be created within azure by using Terraform or ARM templates. Such alerts can be source controlled and be deployed as part of pipelines(Azure Devops pipelines, Jenkins, Github Actions etc.). Some reference of how to do this are: [Terraform Monitor Metric Alert](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/monitor_metric_alert). Alerts can also be created based on log analytics query and can be defined as code using [Terraform Monitor Scheduled Query Rules Alert](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/monitor_scheduled_query_rules_alert#example-usage).
```
Comment (NarmathaBala) [inline observability/readme.md:69]:
```suggestion
2. Alerts as Code - Alerts can be created within azure by using Terraform or ARM templates. Such alerts can be source controlled and be deployed as part of pipelines (Azure DevOps pipelines, Jenkins, Github Actions etc.). Few references of how to do this are: [Terraform Monitor Metric Alert](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/monitor_metric_alert). Alerts can also be created based on log analytics query and can be defined as code using [Terraform Monitor Scheduled Query Rules Alert](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/monitor_scheduled_query_rules_alert#example-usage).
```

File: observability/readme.md [hunk 3]
```diff
@@ -62,6 +62,14 @@ contributions.
...
+
+3. Automating Log Analytics Queries- There are several use cases where automation of log analytics queries may be needed. Example, Automatic Report Generation, Running custom queries programmatically for analysis, debugging etc. For these use cases to work, log queries should be source controlled and automation can be built using [log analytics REST](https://dev.loganalytics.io/documentation/Using-the-API) or [azure cli](https://docs.microsoft.com/en-us/cli/azure/ext/log-analytics/monitor/log-analytics?view=azure-cli-latest).
```
Comment (NarmathaBala) [inline observability/readme.md:71]: nit: space after Queries

**Review by NarmathaBala — CHANGES_REQUESTED** [review]
Lakhsay, few minor updates. Otherwise looking good. Thanks for your contribution

File: observability/readme.md [hunk 4]
```diff
@@ -62,6 +62,14 @@ contributions.
 
 Read more [here](pitfalls.md) to understand what to watch out for while designing and building an observable system.
 
+## Examples of Observability As Code
```
[the remark this answers is not in the document]
Comment (lakshaykaushik) [reply observability/readme.md]: done

File: observability/readme.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (lakshaykaushik) [reply observability/readme.md]: done

File: observability/readme.md (hunk 2 above)
Comment (lakshaykaushik) [reply observability/readme.md]: done

File: observability/readme.md (hunk 3 above)
Comment (lakshaykaushik) [reply observability/readme.md]: done

File: observability/readme.md (hunk 1 above)
Comment (lakshaykaushik) [reply observability/readme.md]: done

**Review by Guilhermeslucas — APPROVED** [review]

File: observability/readme.md [hunk 5]
```diff
@@ -32,6 +32,14 @@ Read on how Observability can be implemented in [Machine Learning](ml-observabil
...
+
+2. Alerts as Code - Alerts can be created within azure by using Terraform or ARM templates. Such alerts can be source controlled and be deployed as part of pipelines (Azure DevOps pipelines, Jenkins, Github Actions etc.). Few references of how to do this are: [Terraform Monitor Metric Alert](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/monitor_metric_alert). Alerts can also be created based on log analytics query and can be defined as code using [Terraform Monitor Scheduled Query Rules Alert](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/monitor_scheduled_query_rules_alert#example-usage).
```
Comment (NarmathaBala) [inline observability/readme.md:39]: nit: Azure (not azure)

**Review by NarmathaBala — APPROVED** [review]
One minor suggestion. Otherwise, this LGTM. Thanks for your contribution @lakshaykaushik

**Pull request merged**